Create 2025-05-14.md (#731)
solid/specification: meetings/2025-05-14.md

* No audio or video recording, or automated transcripts without consent. Meetings are transcribed and made public. If consent is withheld by anyone, recording/retention must not occur.
* Join queue to talk.
* Topics can be proposed at the bottom of the agenda to be discussed as time allows. Make it known if a topic is urgent or cannot be postponed.

### Participation and Code of Conduct
* [Join the W3C Solid Community Group](https://www.w3.org/community/solid/join), [W3C Account Request](http://www.w3.org/accounts/request), [W3C Community Contributor License Agreement](https://www.w3.org/community/about/agreements/cla/)
* [Solid Code of Conduct](https://github.com/solid/process/blob/main/code-of-conduct.md), [Positive Work Environment at W3C: Code of Conduct](https://www.w3.org/policies/code-of-conduct/)
* Operating principle for effective participation is to allow access across disabilities, across country borders, and across time. Feedback on tooling and meeting timing is welcome.
* If this is your first time, welcome! please introduce yourself.

### Scribes
* https://fireflies.ai/

> ---

Author: ```suggestion

### Links
* https://rallly.co/invite/3y8dd4bwJ56C
* https://github.com/w3c-cg/awesome-semantic-shapes?tab=readme-ov-file#shape-convertors-and-generators
* https://shacl-play.sparna.fr/play/convert

---
```

Overview

    Meeting Scheduling Issues: Low attendance linked to current meeting time; a scheduling poll is created to optimize weekly meeting slot. Manual attendance required while AI tool is being tested for notes.
    SAI Playground Demo: Elf Pavlik showcased early SAI playground version for visualizing data grants and delegation. Feedback on usability and clarity for authorization flows requested; emphasis on improving representation of delegation chains.
    Delegation & Policy Enforcement: Discussed access grants vs. third-party sharing limits. Need for various implementation approaches before standardization. Agreed on necessity of technical enforcement and potential legal frameworks.

>     Linked Web Storage Progress: Led by Hadrian with Christoph Braun as a spec editor. Preference for adding use case variants to existing scenarios rather than creating new ones. Suggestions to document subtleties via GitHub.

Reviewer: ```suggestion
    Linked Web Storage Progress: Led by Hadrian with Erich Bremer getting more involved, potentially as a spec editor. Preference for adding use case variants to existing scenarios rather than creating new ones. Suggestions to document subtleties via GitHub.
```

    Low attendance attributed to current meeting time; poll aims to resolve this.
    Manual attendance requested while AI note-taking is being trialed.

‍ SAI Playground Demo and Feedback (07:29 - 15:01)

    Elf Pavlik demonstrated an early version of the SAI playground for visualizing delegation and data grants among social agents and applications.
    Demo illustrated delegation chains, data grants, and the importance of visualizing authorization flows.
    Feedback requested on usability and suggestions for improving clarity, especially in representing delegation chains and policy enforcement limitations.

Delegation, Authorization, and Policy Enforcement (15:01 - 20:33)

    Discussion on whether access grants can restrict third-party sharing and how delegation limits (number of hops) can be enforced or circumvented.
    Implementation details (e.g., warnings, enforcement) left to individual applications; technical enforcement is limited, but policy and legal frameworks may help.
    Consensus that multiple approaches should be explored before standardizing delegation mechanisms.

Linked Web Storage Use Cases and Variants (20:35 - 28:29)

>     Recent progress in Linked Web Storage use cases; Hadrian leading, Christoph Braun becoming more involved as a spec editor.

Author: ```suggestion
    Recent progress in Linked Web Storage use cases; Hadrian leading, Erich Bremer becoming more involved as a spec editor.
```

    Reminder of the Solid Practitioners meeting the next day, featuring Christoph Braun's presentation on delegation.
    Open call for agenda suggestions for the next CG meeting; Hadrian to prepare the agenda.
    General encouragement for ongoing feedback and participation.

Action items
Unassigned

    Review and provide feedback on the scheduling poll for finding the best CG weekly meeting day and time (06:06)
    Provide feedback on the SAI playground demo, especially regarding visualization and usability for understanding delegation and data grants (14:26)
    Explore and possibly bootstrap calendar and event shapes for Solid using existing vocabularies (iCal, RDFcal) and tooling (ShaclPlay, altashackle) (33:22)
    Add yourself to the meeting attendance list manually while AI note-taking is being tested (33:22)

elf Pavlik

    Send the finalized scheduling poll to the mailing list and invite wider participation (06:06)
    Link the GitHub issue regarding delegation limitations and share with Christoph Braun (16:09)
    Contribute use case variants or subtleties to the Linked Web Storage use cases, either by commenting on existing issues or linking related issues (28:29)

>     Coordinate with Hadrian and others to propose a project board or systematic way to track community development priorities and funding opportunities (48:10)

Reviewer: ```suggestion
    Coordinate with Hadrian and others, including Jesse and the ODI, to propose a project board or systematic way to track community development priorities and funding opportunities (48:10)
```

 If we are happy, we can like figure it out how to either get some kind of subscription or actually they. If you. If you share your calendar with them and allow the AI to join all your meetings, then you kind of can use it unlimited. So maybe I just create dedicated account for like solid CG Turing and yeah, we just have a dedicated calendar so you can join all the solid CG events. We'll see how it works. Maybe. Let me ping people that the meeting started. So Christoph, with what I suggested last week that I think would be great if you could present in more detail your work on proxies. Tomorrow there is practitioners meeting. So I will ask everyone if they would like to hold it there because it's more attended than the CG weekly. As you can see, it's not so many people. Hopefully more will join. But what do you think about if practitioners agree to aim two weeks from. No way is it two weeks. Let me check.

Christoph Braun (KIT)  uvdsl: 03:52 
 Actually, I talked to Jeff already and he put me on the schedule for tomorrow.

elf Pavlik: 03:56 
 Oh, for tomorrow? Awesome. Yep. He didn't announce it I think but it would be nice surprise.

Christoph Braun (KIT)  uvdsl: 04:04 

>  Yeah, I think he. He put me on the schedule like that was his message from. Let me double check. Yes, I'm scheduled for May 15, 15 UTC.

Reviewer: ```suggestion
 Yeah, I think he put me on the schedule like that was his message. Let me double check. Yes, I'm scheduled for May 15, 15 UTC.
```

Christoph Braun (KIT)  uvdsl: 15:01 

>  Yeah, thank you, Pavlik. So I think the visualization definitely will.

Reviewer: ```suggestion
 Yeah, thank you, Pavlik. So I think the visualization definitely will help people to understand the topic.
```

elf Pavlik: 15:08 
 Help.

Christoph Braun (KIT)  uvdsl: 15:11 

>  Understanding how like the. Which resources in particular are given access. I think the. From a visual standpoint, maybe you could manage to make it a bit clearer that there is this chain of authorizations, that these really depend on each other. This was the one thing I was a bit missing from the status quo. But again I think it's a good, good aid definitely. I'd have a question regarding the delegation process. Can a. If, if an access grant or an authorization access authorization is given, can a party decide not to enable third party sharing? So like Acme disables that Alice can delegate to Bob or whatever. Did you think of that?

Reviewer: ```suggestion
I think from a visual standpoint, maybe you could manage to make it a bit clearer that there is this chain of authorizations, that these really depend on each other. This was the one thing I was a bit missing from the status quo. But again I think it's a good, good aid definitely. I'd have a question regarding the delegation process. If an access grant or an authorization access authorization is given, can a party decide not to enable third party sharing? So like Acme disables that Alice can delegate to Bob or whatever. Did you think of that?
```

> Christoph Braun (KIT)  uvdsl: 22:25 

Author: ```suggestion
Erich Bremer: 22:25 
```

> Christoph Braun (KIT)  uvdsl: 24:31 

Reviewer: ```suggestion
Erich Bremer: 24:31 
```

 I'm not certain where they're gonna go right now. I'm still, like I said, I just began looking through the existing use cases as they are to try to get my head around them all. My personal preference is I'd rather hear any and all use cases, even if they're new ones, even if they aren't part of that current pool. It's considering everything that is of interest to make sure that we have the foundations for something that's extensible, that doesn't require a rewrite.

elf Pavlik: 25:01 

>  Go ahead, Chris. Go ahead, Christoph.

Reviewer: ```suggestion
Go ahead, Christoph.
```

Christoph Braun (KIT)  uvdsl: 25:04 

>  Yeah, I agree with that. And I, so I'd agree with that. And it makes the subtleties apparent or more obvious because I think some people from the community, they're, they're quite clear on that we need this distinction between the authorization app or agent or whatever mechanism there is to, to authorize access, that this should be distinct from some consuming business application or personal application. Right. So there, there is this notion that people are aware of and I believe these people, me included, will like fight for that. But still, if there are any use cases you can give us as ammunition, so to speak, that would be great.

Reviewer: ```suggestion
 Yeah, I agree with that. And it makes the subtleties apparent or more obvious because I think some people from the community, they're quite clear on that we need this distinction between the authorization app or agent or whatever mechanism there is to authorize access, that this should be distinct from some consuming business application or personal application. Right. So there, there is this notion that people are aware of and I believe these people, me included, will like fight for that. But still, if there are any use cases you can give us as ammunition, so to speak, that would be great.
```

elf Pavlik: 25:59 
 Yeah. So my main dilemma here is that I would want to introduce completely new use case, building a new scenario or maybe work with variants to say, okay, we have the existing use case, let's stick with that, but add a variant to it to say, but it's the same situation, but there's additional kind of requirement of saying actually Alice, don't want to just. Or whatever. Ginion or seven of hearts, whatever names we are using that they don't want to disclose the whole social graft application. So instead of introducing completely new use case, my preference would be just to have a use case and then like some variants of those subtleties, but not like completely separate use cases that someone has to imagine the whole scenario from scratch. And then maybe because of those differences in the scenarios, like different names, different whatever, we would like get distracted from what actually differs in those different variants. And for me would be important how we can actually work with those variants. Say, okay, the whole thing is the same, but there's one variation on that.

Christoph Braun (KIT)  uvdsl: 27:13 

>  I mean from just an operational standpoint, either comment on the existing issue or if you can. I'm not sure how that works on GitHub actually I've never used that feature. Open a sub issue or create a new issue, link the old issue, say everything is the same except for these things. I think then that would also work except Eric has a different opinion on that. I'm sorry, follow your point. So I just was listing ways that would accommodate like public's idea of reusing an existing use case as a basis and then introducing subtleties or like more details, more restrictions on the use case. And I was just saying, well, link the old, link the original use case or comment in that use case with that variant and then people will be able to find that variant and pick up the requirements from there. That was my suggestion. Yeah, I, I would agree with that.

Reviewer: ```suggestion
 I mean from just an operational standpoint, either comment on the existing issue or if you can. I'm not sure how that works on GitHub actually I've never used that feature: open a sub issue. Or create a new issue, link the old issue, say everything is the same except for these things. I think then that would also work except Erich has a different opinion on that. 
 
 Erich Bremer:
 I'm sorry, follow your point?
 
 Christoph Braun (KIT) - uvdsl:
 So I just was listing ways that would accommodate like public's idea of reusing an existing use case as a basis and then introducing subtleties or like more details, more restrictions on the use case. And I was just saying, well, link the old, link the original use case or comment in that use case with that variant and then people will be able to find that variant and pick up the requirements from there. That was my suggestion. 
 
 Erich Bremer:
 Yeah, I would agree with that.
```

Marc Haddle: 30:45 
 Calendars and events are not new to the semantic web world. Are you basing these shapes off? Are they shapes of existing vocabularies that you're thinking? If so, what vocabularies are you looking at?

elf Pavlik: 31:01 
 Yes, I think the most established standard is I calendar that it's One of the RFCs I had linked last week. It's in a minute and I think that there's also RDF cal which I understand it's like mapping to rdf so that will be probably my first direction. So it's easy to consume the ICAL feeds and events which I think all of us use in our day to day coordination and have an easy way of importing them and also whatever we have in the storage to have a way to transform and export it to iCal. So that would be my preference to take it from this angle and as soon as possible bridge maybe the solid CG calendar make it available as RDF and as we start scheduling meetings with one another we also start eventually maybe in few months using the solid and still be able to export it to our Google Calendars, Outlook, whatever people are using.

> Marc Haddle: 32:02 

Author: ```suggestion
Jesse Wright: 32:02 
```

> Marc Haddle: 39:56 

Reviewer: ```suggestion
Jesse Wright: 39:56 
```

Marc Haddle: 44:14 
 To respond to Kristoff first because it's in memory. We need to think about this better. Elf and I this might actually be a good conversation for us to have one on one as to what you'd like this to look like on the CG side and maybe bring Jeff in as well to see how we can take what's in the catalog because there's some stuff in the catalogue around funding because really what we need to track here are funding opportunities, interest in development and functional requirements that people are after. So we do need to think about that. And that's a good point. Christophe. Theo, I was going to respond to something you said but I've now forgotten it. Could you remind me what your statement was?

Theo @thhck: 45:08 
 Sure, yeah. It was that we can make a component to CSS to make people use their own domain name, but we can also work at the hostel level. And that reminds me that at last post them I meet like in the hoster. I think.

> Marc Haddle: 45:36 

Reviewer: ```suggestion
Jesse Wright: 45:36 
```

 You'Ve reminded me now if those. So it would. It would be good if they are interested in hosting things. Can you either like just so we action things rather than talking about it, can you reach out to them and actually see if they're interested? If they are, we can then have conversations to work out how to support them in getting the infrastructure set up or if you don't have the bandwidth to do that, put me in contact and I can start to have those conversations.

Theo @thhck: 46:11 
 No, no, I can, I can reach out and, and see.

elf Pavlik: 46:16 
 Yeah.

Theo @thhck: 46:16 
 And take the temperature and then maybe we can have a meeting altogether.

> Marc Haddle: 46:20 

Author: ```suggestion
Jesse Wright: 46:20 
```

Christoph Braun (KIT)  uvdsl: 53:05 

>  Thank you, Pac. All right, bye, everybody.

Reviewer: ```suggestion
 Thank you, Pavlik. All right, bye, everybody.
```
